DOWNLOAD_ONLY not parsed?

I have the following setup, with _NEWLIB_ and its dependency _jegp_ both depending on _range-v3_.
Having translated directly from a sequence of `FetchContent`s and accompanying `FetchContent_MakeAvailable` to a sequence of `CPMFindPackage`, _jegp_ is first found and finds _range-v3_, but when _NEWLIB_ does it, `DOWNLOAD_ONLY` is `""`, as seen in the error below, for which I used [`cmake_print_variables`](https://cmake.org/cmake/help/latest/module/CMakePrintHelpers.html).

```
Running /usr/bin/cmake -GNinja <source-root>/newlib in <source-root>/newlib/build.
-- Found SFML 2.5.1 in /usr/lib64/cmake/SFML
-- CPM: adding package jegp_cmake_modules@0 (master)
-- CPM: adding package jegp@0 (master)
-- CMAKE_CURRENT_LIST_FILE="<source-root>/newlib/build/_deps/jegp-src/cmake/JEGPFindDependencies.cmake" ; CPM_ARGS_NAME="jegp_cmake_modules" ; DOWNLOAD_ONLY="NO"
-- CPM: jegp: adding package range-v3@0 (master)
-- CMAKE_CURRENT_LIST_FILE="<source-root>/newlib/tools/cmake/NEWLIBFindDependencies.cmake" ; CPM_ARGS_NAME="range-v3" ; DOWNLOAD_ONLY=""
CMake Error at tools/cmake/CPM.cmake:167 (cpm_fetch_package):
  cpm_fetch_package Function invoked with incorrect arguments for function
  named: cpm_fetch_package
Call Stack (most recent call first):
  tools/cmake/CPM.cmake:135 (CPMCheckIfPackageAlreadyAdded)
  tools/cmake/NEWLIBFindDependencies.cmake:21 (CPMFindPackage)
  CMakeLists.txt:11 (include)


-- CPM: using local package fmt@0
-- CPM: adding package mp-units@0 (master)
-- Configuring incomplete, errors occurred!
See also "<source-root>/newlib/build/CMakeFiles/CMakeOutput.log".
CMake process exited with exit code 1.
Elapsed time: 00:03.
```

I have circumvented this by first declaring my direct dependencies for which I know my other direct dependencies depend upon, which is necessary since the content is immediately made available, but it could become a nightmare in more complex scenarios.

--- a/cmake/CPM.cmake
+++ b/cmake/CPM.cmake
@@ -28,11 +28,11 @@
 
 cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
 
-set(CURRENT_CPM_VERSION 0.23)
+set(CURRENT_CPM_VERSION 0.24)
 
 if(CPM_DIRECTORY)
-  if(NOT ${CPM_DIRECTORY} STREQUAL ${CMAKE_CURRENT_LIST_DIR})
-    if (${CPM_VERSION} VERSION_LESS ${CURRENT_CPM_VERSION})
+  if(NOT CPM_DIRECTORY STREQUAL CMAKE_CURRENT_LIST_DIR)
+    if (CPM_VERSION VERSION_LESS CURRENT_CPM_VERSION)
       message(AUTHOR_WARNING "${CPM_INDENT} \
 A dependency is using a more recent CPM version (${CURRENT_CPM_VERSION}) than the current project (${CPM_VERSION}). \
 It is recommended to upgrade CPM to the most recent version. \
@@ -162,10 +162,7 @@ function(CPMCheckIfPackageAlreadyAdded CPM_ARGS_NAME CPM_ARGS_VERSION CPM_ARGS_O
         endif()
       endforeach()
     endif()
-    cpm_fetch_package(${CPM_ARGS_NAME} ${DOWNLOAD_ONLY})
     cpm_get_fetch_properties(${CPM_ARGS_NAME})
-    SET(${CPM_ARGS_NAME}_SOURCE_DIR "${${CPM_ARGS_NAME}_SOURCE_DIR}")
-    SET(${CPM_ARGS_NAME}_BINARY_DIR "${${CPM_ARGS_NAME}_BINARY_DIR}")  
     SET(${CPM_ARGS_NAME}_ADDED NO)
     SET(CPM_PACKAGE_ALREADY_ADDED YES PARENT_SCOPE)
     cpm_export_variables()

